jdaviz/core/template_mixin.py

```diff
@@ -281,24 +281,42 @@ def hub(self):
     def viewer_dicts(self):
         def _dict_from_viewer(viewer, viewer_item):
             d = {'viewer': viewer,
-                 'id': viewer_item['id'],
-                 'icon': self.app.state.viewer_icons.get(viewer_item['id'])}
+                 'id': viewer_item.get('id'),
+                 'icon': self.app.state.viewer_icons.get(viewer_item.get('id'))}
             if viewer_item.get('reference') is not None:
-                d['reference'] = viewer_item['reference']
-                d['label'] = viewer_item['reference']
+                d['reference'] = viewer_item.get('reference')
+                d['label'] = viewer_item.get('reference')
             else:
                 d['reference'] = None
-                d['label'] = viewer_item['id']
+                d['label'] = viewer_item.get('id')
 
             return d
 
         return [_dict_from_viewer(viewer, self.app._viewer_item_by_id(vid))
                 for vid, viewer in self.app._viewer_store.items()
                 if viewer.__class__.__name__ != 'MosvizTableViewer']
 
-    @cached_property
+    @property
     def spectrum_viewer(self):
-        return self._plugin.app.get_viewer("spectrum-viewer")
+        if hasattr(self, '_default_spectrum_viewer_reference_name'):
+            viewer_reference = self._default_spectrum_viewer_reference_name
+        else:
+            viewer_reference = self.app.get_first_viewer_reference_name(
+                require_spectrum_viewer=True
+            )
+
+        return self._plugin.app.get_viewer(viewer_reference)
```

Comment:
I wonder if this should be abstracted away in the app somewhere or be a part of the call to `get_first_viewer_reference_name(use_default=True)` or something similar?

Especially since this logic is then repeated at least once more below for `spectrum_2d_viewer`.